tests/baselines/reference/indexedAccessRetainsIndexSignature.js

```diff
@@ -0,0 +1,10 @@
+//// [indexedAccessRetainsIndexSignature.ts]
+type Diff<T extends string, U extends string> =
+    ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T]
+type Omit<U, K extends keyof U> = Pick<U, Diff<keyof U, K>>
```

Comment:
(on many people's wishlist): add these to `lib.d.ts` 💟

Reply:
Diff and Omit might someday be in lib.d.ts once conditional types are in Typescript, but the current implementation is way too hacky.

Reply:
In fact, only Diff implementation is way too hacky. But it will be solved if subtraction types make it into typescript (someday).